packages/web-app-files/src/helpers/resource/actions/upload.ts

```diff
@@ -0,0 +1,244 @@
+
+      if (doForAllConflicts && !isFolder) {
+        conflictArray.push({
+          name: conflict.name,
+          strategy: allConflictsStrategy
+        })
+        continue
+      }
+      if (doForAllConflictsFolders && isFolder) {
+        conflictArray.push({
+          name: conflict.name,
+          strategy: allConflictsStrategyFolders
+        })
+        continue
+      }
+
+      const resolvedConflict: ResolveConflict = await this.resolveFileExists(
+        { name: conflict.name, isFolder } as Resource,
+        allConflictsCount - count,
+        false,
+        isFolder
+      )
+      count++
+      if (resolvedConflict.doForAllConflicts) {
+        if (isFolder) {
+          doForAllConflictsFolders = true
+          allConflictsStrategyFolders = resolvedConflict.strategy
+        } else {
+          doForAllConflicts = true
+          allConflictsStrategy = resolvedConflict.strategy
+        }
+      }
+
+      conflictArray.push({
+        name: conflict.name,
+        strategy: resolvedConflict.strategy
+      })
+    }
+    const filesToSkip = resolvedFileConflicts
+      .filter((e) => e.strategy === ResolveStrategy.SKIP)
+      .map((e) => e.name)
+    const foldersToSkip = resolvedFolderConflicts
+      .filter((e) => e.strategy === ResolveStrategy.SKIP)
+      .map((e) => e.name)
+
+    files = files.filter((e) => !filesToSkip.includes(e.name))
+    files = files.filter(
+      (file) =>
+        !foldersToSkip.some((folderName) => file.meta.relativeFolder.split('/')[1] === folderName)
+    )
+
+    const filesToKeepBoth = resolvedFileConflicts
+      .filter((e) => e.strategy === ResolveStrategy.KEEP_BOTH)
+      .map((e) => e.name)
+    const foldersToKeepBoth = resolvedFolderConflicts
+      .filter((e) => e.strategy === ResolveStrategy.KEEP_BOTH)
+      .map((e) => e.name)
+
+    for (const fileName of filesToKeepBoth) {
+      const file = files.find((e) => e.name === fileName && !e.meta.relativeFolder)
+      const extension = extractExtensionFromFile({ name: fileName } as Resource)
+      file.name = resolveFileNameDuplicate(fileName, extension, this.currentFiles)
+    }
+    for (const folder of foldersToKeepBoth) {
+      const filesInFolder = files.filter((e) => e.meta.relativeFolder.split('/')[1] === folder)
+      for (const file of filesInFolder) {
+        const newFolderName = resolveFileNameDuplicate(folder, '', this.currentFiles)
+        file.meta.relativeFolder = file.meta.relativeFolder.replace(
+          `/${folder}`,
+          `/${newFolderName}`
+        )
+        file.meta.relativePath = file.meta.relativePath.replace(`/${folder}/`, `/${newFolderName}/`)
+        file.meta.tusEndpoint = file.meta.tusEndpoint.replace(`/${folder}`, `/${newFolderName}`)
+        const data = file.data as any
+        data.relativePath = data.relativePath.replace(`/${folder}/`, `/${newFolderName}/`)
+      }
+    }
+    if (files.length === 0) return
```

Comment:
Please adjust these lines to: https://github.com/owncloud/web/pull/7724/files#diff-4211e1c96725b1778d5fbab11cda641ebd5c4416ffde9c72ffe091ecd77f224a

This fixes several bugs we had with the overlay.

Reply:
thx will do it, I think I overlooked it at rebasing